src/UIKit/UITraitChangeObservable.cs

```diff
@@ -0,0 +1,311 @@
+			where T2 : IUITraitDefinition
+			where T3 : IUITraitDefinition
+		{
+			return RegisterForTraitChanges (ToClasses (typeof (T1), typeof (T2), typeof (T3)), handler);
+		}
+
+		/// <summary>
+		/// Registers a callback handler that will be executed when any of the specified traits changes.
+		/// </summary>
+		/// <typeparam name="T1">A trait to observe</typeparam>
+		/// <typeparam name="T2">A trait to observe</typeparam>
+		/// <typeparam name="T3">A trait to observe</typeparam>
+		/// <typeparam name="T4">A trait to observe</typeparam>
+		/// <param name="handler">The callback to execute when any of the specified traits changes.</param>
+		/// <returns>A token that can be used to unregister the callback by calling <see cref="M:UnregisterForTraitChanges" />.</returns>
+		public unsafe IUITraitChangeRegistration RegisterForTraitChanges<T1, T2, T3, T4> (Action<IUITraitEnvironment, UITraitCollection> handler)
+			where T1 : IUITraitDefinition
+			where T2 : IUITraitDefinition
+			where T3 : IUITraitDefinition
+			where T4 : IUITraitDefinition
+		{
+			return RegisterForTraitChanges (ToClasses (typeof (T1), typeof (T2), typeof (T3), typeof (T4)), handler);
+		}
+
+		/// <summary>
+		/// Registers a selector that will be called on the specified object when any of the specified traits changes.
+		/// </summary>
+		/// <param name="traits">The traits to observe.</param>
+		/// <param name="target">The object whose specified selector will be called.</param>
+		/// <param name="action">The selector to call on the specified object.</param>
+		/// <returns>A token that can be used to unregister the callback by calling <see cref="M:UnregisterForTraitChanges" />.</returns>
+		public IUITraitChangeRegistration RegisterForTraitChanges (Type [] traits, NSObject target, Selector action)
+		{
+			return RegisterForTraitChanges (ToClasses (traits), target, action);
+		}
+
+		/// <summary>
+		/// Registers a selector that will be called on the current object when any of the specified traits changes.
+		/// </summary>
+		/// <param name="traits">The traits to observe.</param>
+		/// <param name="action">The selector to call on the current object.</param>
+		/// <returns>A token that can be used to unregister the callback by calling <see cref="M:UnregisterForTraitChanges" />.</returns>
+		public IUITraitChangeRegistration RegisterForTraitChanges (Type [] traits, Selector action)
+		{
+			return RegisterForTraitChanges (ToClasses (traits), action);
+		}
+
+
+#if !XAMCORE_5_0
+		[BindingImpl (BindingImplOptions.Optimizable)]
+#if !NET
+		public IUITraitChangeRegistration RegisterForTraitChanges (Class [] traits, global::System.Action<IUITraitEnvironment, UITraitCollection> handler)
+		{
+			// The manual block code is somewhat annoying to implement, so at least don't do it twice (once for .NET and once for legacy Xamarin) unless we really need to.
+			throw new NotImplementedException ("This API has not been implemented for legacy Xamarin. Please upgrade to .NET");
```

Comment:
Do we want this translated? Specially when it tells to upgrade.

Reply:
We don't translate anything else in our platform assemblies, so we'd first have to implement that part (which isn't trivial - we'd even have to figure out how to ship translations), and if we do this, it should be implemented in a separate PR.